backend/api.py

```
@@ -2927,6 +2991,627 @@ def find_similar_ligands():
+    against that — so the feature works for any ChEMBL drug, not just the
+    pre-ingested subset.
+
+    Each result row's `name` is enriched with ChEMBL `pref_name` when a
+    chembl_id is present, so the UI shows recognizable names ("ANASTROZOLE")
+    rather than IUPAC strings from our local molecules table.
+    """
+    try:
+        top_k = max(1, min(int(request.args.get('top_k', 20)), 100))
+        engine = get_engine()
+
+        results: List[Dict] = []
```

Ruff (0.15.12)

[error] Undefined name `List`

(F821)